cli/listener/src/main.rs

```diff
@@ -392,7 +403,154 @@ impl Metrics {
+/// `input`. `input` is the url query in bytes.
+fn get_date_params(parsed_inputs: Vec<(Cow<str>, Cow<str>)>) -> Result<DateFromTo, ResponseError> {
+    let (first_k, first_v) = parsed_inputs.get(0).ok_or(ResponseError {
+        error: "First parameter \"from\" or \"to\" not provided.".to_owned(),
+    })?;
+    let (second_k, second_v) = parsed_inputs.get(1).ok_or(ResponseError {
+        error: "Second parameter \"from\" or \"to\" not provided.".to_owned(),
+    })?;
+    match (
+        (first_k.as_ref(), first_v.as_ref()),
+        (second_k.as_ref(), second_v.as_ref()),
+    ) {
+        (("from", from), ("to", to)) | (("to", to), ("from", from)) => {
+            Ok(parse_utc_iso8601(from)..parse_utc_iso8601(to))
+        }
+        _ => Err(ResponseError {
+            error: "Wrong parameters provided, query parameters should be \"from\" and \"to\"."
+                .to_owned(),
+        }),
+    }
+}
+
+/// Returns a Request response with an error.
+/// With the status code set to 400.
+fn get_error_response(err_res: ResponseError) -> ResponseBox {
+    let content_type = Header::from_bytes(
+        &b"Content-Type"[..],
+        &b"application/json; charset=UTF-8"[..],
+    )
+    .expect("Static header value, does not fail at runtime.");
+    Response::from_data(serde_json::to_vec(&err_res).expect("Serialization shouldn't fail"))
+        .with_status_code(400)
+        .with_header(content_type)
+        .boxed()
+}
+
+/// Get an interval price, consume it and returns a `ResponseBox` with the
+/// provided interval price and computed annual percentage rate.
+fn get_success_response(interval_prices: IntervalPrices) -> ResponseBox {
+    let response_interval = ResponseInterval {
+        annual_percentage_rate: interval_prices.annual_percentage_rate(),
+        interval_prices,
+    };
+    let content_type = Header::from_bytes(&b"Content-Type"[..], &b"application/json"[..])
+        .expect("Static header value, does not fail at runtime.");
+    Response::from_data(
+        serde_json::to_vec(&response_interval).expect("Serialization shouldn't fail"),
+    )
+    .with_header(content_type)
+    .boxed()
+}
+
+/// Gets a response that will be sent to a requesting client for a price
+/// interval. Returns None if no method is passed.
+fn get_interval_price_request(
+    db_connection: &Connection,
+    request_url: &str,
+) -> Option<ResponseBox> {
+    let parsed_url = match Url::parse(request_url) {
+        Ok(parsed_url) => parsed_url,
+        Err(err) => {
+            return Some(get_error_response(ResponseError {
+                error: err.to_string(),
+            }))
+        }
+    };
+    let method_name = parsed_url.path_segments()?.last()?;
+    if method_name != "interval_price" {
+        return Some(get_error_response(ResponseError {
+                        error: format!("Method not supported: {}, use \"/interval_price?from=<from_date_iso8601>&to=<to_date_iso8601>\"", method_name)
```

Comment:
Also we now use `from` and `to` here, but `begin` and `end` in the response. We should be consistent, let’s rename these to `begin` and `end`.